File: Python/Challenge 5.py
# Challenge 5
# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

# Step by 2520 as it's the smallest number divisible by 1:10
i = 2520

solution = False

# Only need to check divisible by 11:20 as any number will also be divisble by 1:10
while  solution == False:
    if i % 11 == 0 and i % 12 == 0 and i % 13 == 0 and i % 14 == 0 and i % 15 == 0 and i % 16 == 0 and i % 17 == 0 and i % 18 == 0 and i % 19 == 0 and i % 20 == 0:
        solution = True
        print(i)
    i += 2520

# Result is 232,792,560

File: R/Challenge 5.R
# Challenge 5
# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

#Check Function
divide_function <- function(i) {
  for(j in 11:20){
    if(i %% j == 0){
      result <- TRUE
    } else {
      result <- FALSE
      break
    }
  }
  print(result)
}
  
#Setup: step of 2520 as it is the smallest number that is divisible by 1:10
step <- 2520

i <- 2520

solution <-  FALSE

#Loop
while(solution == FALSE){
  if(divide_function(i) == TRUE) {
    solution <-  TRUE
    break
    print(i)
  } else {
    i <- i + step
  }
}

#Answer is 232,792,560
